File: KnowledgeCenterServer/_Match/KnowledgeCenter.Match.Contracts/CustomerOfferSkill.cs
using System.ComponentModel.DataAnnotations;
using System;

namespace KnowledgeCenter.Match.Contracts
{
    public class CustomerOfferSkill
    {
        public int Id { get; set; }

        [Required]
        public int SkillId { get; set; }
        public Skill Skill { get; set; }
        
        [Required]
        public int SkillLevelId { get; set; }
        public SkillLevel SkillLevel { get; set; }
        
        [Required]
        public int SkillPriority { get; set; }
    }
}

File: KnowledgeCenterClient/src/app/match/models/CustomerOfferSkill.ts
import {Skill} from './Skill';
import {SkillLevel} from './SkillLevel';

export interface CustomerOfferSkill {
  id: number;
  skillId: number;
  skill: Skill;
  skillLevelId: number;
  skillLevel: SkillLevel;
  skillPriority: number;
  creationDate: Date;
  modificationDate: Date;
}
